src/main/java/io/xpring/ClassicAddress.java

```diff
@@ -0,0 +1,24 @@
+package io.xpring;
+
+import io.xpring.javascript.JavaScriptWallet;
+import io.xpring.javascript.JavaScriptWalletFactory;
+import io.xpring.javascript.JavaScriptWalletGenerationResult;
+import org.immutables.value.Value;
+
+import java.util.Optional;
+
+/**
+ * Represents classic address components on the XRP Ledger.
+ */
+@Value.Immutable
+public abstract class ClassicAddress {
```

Comment:
For future: using an `abstract class` is just fine, and equivalent to what I'm about to say, but using Java interface's with Immutables requires less boilerplate code. E.g., here's the interface equivalent:

```
/**
 * Represents classic address components on the XRP Ledger.
 */
@Immutable
public interface ClassicAddress {
/**
     * The address component of the classic address.
     */
    String address();

    /**
     * The tag component of the classic address.
     */
    Optional<Long> tag();
}
```

Reply:
Oh interesting. Their tutorial uses `abstract class`: 
https://immutables.github.io/getstarted.html

I am happier with `interface` anyway so I'll make this change.

Reply:
Oh interesting. Their tutorial uses `abstract class`: 
https://immutables.github.io/getstarted.html

I am happier with `interface` anyway so I'll make this change.